executor/builder.go

```diff
@@ -1259,24 +1260,35 @@ func (b *executorBuilder) getStartTS() (uint64, error) {
 
 func (b *executorBuilder) buildMemTable(v *plannercore.PhysicalMemTable) Executor {
 	var e Executor
-	switch v.Table.Name.L {
-	case strings.ToLower(infoschema.TableClusterConfig):
-		e = &ClusterReaderExec{
+	switch v.DBName.L {
+	case util.MetricSchemaLowerName:
+		e = &MetricReaderExec{
 			baseExecutor: newBaseExecutor(b.ctx, v.Schema(), v.ExplainID()),
-			retriever: &clusterConfigRetriever{
-				extractor: v.Extractor.(*plannercore.ClusterConfigTableExtractor),
-			},
+			table:        v.Table,
+			outputCols:   v.Columns,
 		}
-	default:
-		tb, _ := b.is.TableByID(v.Table.ID)
-		e = &TableScanExec{
-			baseExecutor:   newBaseExecutor(b.ctx, v.Schema(), v.ExplainID()),
-			t:              tb,
-			columns:        v.Columns,
-			seekHandle:     math.MinInt64,
-			isVirtualTable: !tb.Type().IsNormalTable(),
+	case util.InformationSchemaLowerName:
+		switch v.Table.Name.L {
+		case strings.ToLower(infoschema.TableClusterConfig):
+			e = &ClusterReaderExec{
+				baseExecutor: newBaseExecutor(b.ctx, v.Schema(), v.ExplainID()),
+				retriever: &clusterConfigRetriever{
+					extractor: v.Extractor.(*plannercore.ClusterConfigTableExtractor),
+				},
+			}
 		}
 	}
+	if e != nil {
```

Comment:
This check is weak (if someday we use a builder to construct the executor and builder will return nil, which will break this assumption), prefers to use the `default` case.

Reply:
done.